pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -861,7 +861,7 @@ func (ssc *defaultStatefulSetControl) inPlaceUpdatePod(
 				ssc.recorder.Eventf(set, v1.EventTypeNormal, "SuccessfulUpdatePodInPlace", "successfully update pod %s in-place(revision %v)", pod.Name, updateRevision.Name)
 				return res.InPlaceUpdate, nil
 			}
-
+			updateExpectations.ExpectUpdated(getStatefulSetKey(set), updateRevision.Name, pod)
```

Comment:
Should it be in the `if res.UpdateErr == nil { ... }` above?

Reply:
Done.  Maybe we should add some unit tests to cover this scenario